fast_matched_filter/fast_matched_filter.py

```diff
@@ -246,21 +271,39 @@ def matched_filter(
         )
 
     # list of arguments
-    args = (
-        templates.ctypes.data_as(ct.POINTER(ct.c_float)),
-        sum_square_templates.ctypes.data_as(ct.POINTER(ct.c_float)),
-        moveouts.ctypes.data_as(ct.POINTER(ct.c_int)),
-        data.ctypes.data_as(ct.POINTER(ct.c_float)),
-        weights.ctypes.data_as(ct.POINTER(ct.c_float)),
-        step,
-        n_samples_template,
-        n_samples_data,
-        n_templates,
-        n_stations,
-        n_components,
-        n_corr,
-        cc.ctypes.data_as(ct.POINTER(ct.c_float)),
-    )
+    if variable_template_length:
+        assert n_samples_template.shape == moveouts.shape
+        args = (
+            templates.ctypes.data_as(ct.POINTER(ct.c_float)),
+            sum_square_templates.ctypes.data_as(ct.POINTER(ct.c_float)),
+            moveouts.ctypes.data_as(ct.POINTER(ct.c_int)),
+            data.ctypes.data_as(ct.POINTER(ct.c_float)),
+            weights.ctypes.data_as(ct.POINTER(ct.c_float)),
+            step,
+            n_samples_template.ctypes.data_as(ct.POINTER(ct.c_int)),
+            n_samples_data,
+            n_templates,
+            n_stations,
+            n_components,
+            n_corr,
+            cc.ctypes.data_as(ct.POINTER(ct.c_float)),
+        )
+    else:
+        args = (
+            templates.ctypes.data_as(ct.POINTER(ct.c_float)),
+            sum_square_templates.ctypes.data_as(ct.POINTER(ct.c_float)),
+            moveouts.ctypes.data_as(ct.POINTER(ct.c_int)),
+            data.ctypes.data_as(ct.POINTER(ct.c_float)),
+            weights.ctypes.data_as(ct.POINTER(ct.c_float)),
+            step,
+            n_samples_template,
+            n_samples_data,
+            n_templates,
+            n_stations,
+            n_components,
+            n_corr,
+            cc.ctypes.data_as(ct.POINTER(ct.c_float)),
+        )
```

Comment:
```suggestion
     args = [
            templates.ctypes.data_as(ct.POINTER(ct.c_float)),
            sum_square_templates.ctypes.data_as(ct.POINTER(ct.c_float)),
            moveouts.ctypes.data_as(ct.POINTER(ct.c_int)),
            data.ctypes.data_as(ct.POINTER(ct.c_float)),
            weights.ctypes.data_as(ct.POINTER(ct.c_float)),
            step,
            n_samples_template,
            n_samples_data,
            n_templates,
            n_stations,
            n_components,
            n_corr,
            cc.ctypes.data_as(ct.POINTER(ct.c_float)),
        ]
    if variable_template_length:
        assert n_samples_template.shape == moveouts.shape
        args[6] = n_samples_template.ctypes.data_as(ct.POINTER(ct.c_int))   
```
Just to be shorter, but no great reason to make this change...